### PR TITLE
Slip refactor

### PR DESCRIPTION
**Pull request by ShiftyRail**
[system] [review nightmare]

## What

Refactors the slipping system.
Previously, you'd have checks dispersed to the four winds, some of them redundant, magic numbers, something I personally find highly anti-OOP in the overlay slipping logic, and lube slipping was hardcoded as heck.

This attemps to simplify the code related to slipping so as to make adding an anti-lube property to shoes easier. This also uses defines instead of magic `-1`.

Posting this "early" to give time for review.

## Testing done (and to do)

- [x] Bananas, soap, bluespess bananas : slip on run, walk, don't slip on (activated) magboots/no slip shoes.
- [x] Water : slip on run, don't slip on walk or (activated) booties.
- [x] Spess : slip on anything that isn't activated magbooties
- [x] Lube : impossible to escape with running, walking, or booties. Not tested crawling or wheelchairs yet.

### CODE REVIEW

**Comment by strawberrytan** [comment]
Wot if noslips could walk on lube

**Comment by ShiftyRail** [comment]
@strawberrytan In a next PR I'll try to push that. The goal of this PR is to make no_lube syndie shoes possible without being horribly snowflake.

**Comment by Dacendeth** [comment]
what about butterslips and clown pda?

**Comment by ShiftyRail** [comment]
They're on the list too @Dacendeth

**Comment by ShiftyRail** [comment]
Tested borgos (and borgs with booties in space), wheelchairs, scooter chairs, butter, clown PDA, and aliens ; all of them work as on live serb (aliens do slip on lube, they're just not stunned.)

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -1805,3 +1802,13 @@ Thanks.
 	// TODO which is pretty irrelevant now but should be fixed
 	tool.reagents.reaction(src, INGEST)
 	return ..()
+
+/mob/living/proc/ApplySlip(var/obj/effect/overlay/puddle/P)
+	return on_foot()
```
Comment (unid15) [inline code/modules/mob/living/living.dm]: Could use a comment explaining that this returns 1 if the slip is possible (which is later checked by ``..()``)

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -502,22 +502,14 @@
...
-	return !locked_to && !lying && !unslippable
-
-/mob/living/proc/Slip(stun_amount, weaken_amount, slip_on_walking = 0)
-	stop_pulling()
-	Stun(stun_amount)
-	Knockdown(weaken_amount)
-	score["slips"]++
-	return 1
+/mob/living/carbon/CheckSlip(slip_on_walking = FALSE, overlay_type = TURF_WET_WATER, slip_on_magbooties = FALSE)
+	var/walking_factor = (!slip_on_walking && m_intent == "walk")
```
Comment (DamianX) [inline code/modules/mob/living/carbon/carbon.dm]: `"walk"` -> `M_INTENT_WALK`

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 3]
```diff
@@ -152,8 +152,21 @@
...
+	var/magboots_slip_factor = (!slip_on_magbooties && shoes_slip_factor && istype(shoes, /obj/item/clothing/shoes/magboots))
 	. = ..()
-	if(. && shoes && shoes.clothing_flags & NOSLIP)
-		. = (istype(shoes, /obj/item/clothing/shoes/magboots) ? -1 : 0)
-	return .
+
+	// We have magboots, and magboots can protect us
+	if (. && magboots_slip_factor)
+		return SLIP_HAS_MAGBOOTIES
+	// We don't have magbboots, or magboots can't protect us
```
Comment (DamianX) [inline code/modules/mob/living/carbon/human/human_movement.dm]: `magbboots` -> `magboots`

**Comment by ShiftyRail** [comment]
Just tested vehicles (wheelchairs and office chairs), works as on live server.

File: __DEFINES/turfs.dm [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
 #define TURF_WET_WATER 1
 #define TURF_WET_LUBE 2
+#define TURF_WET_ICE 3
 
 #define TURF_CONTAINS_REGULAR_ROCKERNAUT 1
-#define TURF_CONTAINS_BOSS_ROCKERNAUT 2
\ No newline at end of file
+#define TURF_CONTAINS_BOSS_ROCKERNAUT 2
+
+#define SLIP_HAS_MAGBOOTIES -1
```
Comment (MadmanMartian) [inline __DEFINES/turfs.dm]: >magbooties

**Pull request merged**